File: injector/src/test/java/org/codejargon/feather/AmbiguousModuleTest.java
package org.codejargon.feather;

import org.junit.Test;

public class AmbiguousModuleTest {
    @Test(expected = InjectorException.class)
    public void ambiguousModule() {
        Injector.with(new Module());
    }

    public static class Module {
        @Provides
        String foo() {
            return "foo";
        }

        @Provides
        String bar() {
            return "bar";
        }
    }
}

File: injector/src/test/kotlin/net/mready/inject/AmbiguousModuleTest.kt
package net.mready.inject

import org.junit.Test

class AmbiguousModuleTest {
    @Test(expected = InjectorException::class)
    fun ambiguousModule() {
        Injector.with(Module())
    }

    class Module {
        @Provides
        fun foo(): String = "foo"

        @Provides
        fun bar(): String = "bar"
    }
}
